das/daser.go

```diff
@@ -68,195 +72,82 @@ func (d *DASer) Start(context.Context) error {
 		return err
 	}
 
-	d.ctx, d.cancel = context.WithCancel(context.Background())
-
 	// load latest DASed checkpoint
-	checkpoint, err := loadCheckpoint(d.ctx, d.cstore)
+	cp, err := loadCheckpoint(ctx, d.cstore)
 	if err != nil {
-		return err
-	}
-	log.Infow("loaded checkpoint", "height", checkpoint)
-
-	// kick off catch-up routine manager
-	go d.catchUpManager(d.ctx, checkpoint)
-	// kick off sampling routine for recently received headers
-	go d.sample(d.ctx, sub, checkpoint)
-	return nil
-}
+		h, err := d.getter.Head(ctx)
+		if err == nil {
+			cp = checkpoint{
+				MinSampled: 1,
+				MaxKnown:   uint64(h.Height),
+			}
 
-// Stop stops sampling.
-func (d *DASer) Stop(ctx context.Context) error {
-	// Stop func can now be invoked twice in one Lifecycle, when:
-	// * BEFP is received;
-	// * node is stopping;
-	// this statement helps avoiding panic on the second Stop.
-	// If ctx.Err is not nil then it means that Stop was called for the first time.
-	// NOTE: we are expecting *only* ContextCancelled error here.
-	if d.ctx.Err() == context.Canceled {
-		return nil
-	}
-	d.cancel()
-	// wait for both sampling routines to exit
-	for i := 0; i < 2; i++ {
-		select {
-		case <-d.catchUpDn:
-		case <-d.sampleDn:
-		case <-ctx.Done():
-			return ctx.Err()
+			log.Warnw("set max known height from store", "height", cp.MaxKnown)
 		}
 	}
-	d.cancel = nil
-	return nil
-}
-
-// sample validates availability for each Header received from header subscription.
-func (d *DASer) sample(ctx context.Context, sub header.Subscription, checkpoint int64) {
-	// indicate sampling routine is running
-	d.indicateRunning()
-	defer func() {
-		sub.Cancel()
-		// indicate sampling routine is stopped
-		d.indicateStopped()
-		// send done signal
-		d.sampleDn <- struct{}{}
-	}()
-
-	// sampleHeight tracks the last successful height of this routine
-	sampleHeight := checkpoint
-
-	for {
-		h, err := sub.NextHeader(ctx)
-		if err != nil {
-			if err == context.Canceled {
-				return
-			}
+	log.Infow("loaded checkpoint", "height", cp)
```

Comment:
```suggestion
	log.Infow("loaded checkpoint", "last_height", cp.MinSampled)
```

Reply:
Or we should make prettier `String()` impl for the checkpoint

Reply:
good idea, done it